apiClients/clientWithTokenApi.ts

```diff
@@ -25,10 +15,16 @@ const clientWithTokenApi = axios.create({
 });
 
 clientWithTokenApi.interceptors.request.use(

```

Comment:
Jest to fajne podejście, ogólnie ludzie sie chyba dzielą na tych którzy robią sprawdzanie tokenu w requestach a inni w responsach.

Chat podpowiada, ze w requescie na plus jest to że natychmiastowo zabezpieczamy request i nie musimy dodatkowo na backendzie obslugiwac tego, plus jest to bezpieczeniejsze dla usera, minusem jest to ze za kazdym requestem to sprawdzamy, dekodujemy token, co przy wieelu zapytaniach moze byc obciazajace.

W responsie natomiast jest na pewno lzej, bo dopiero wtedy gdy backend zwroci ze masz token wygasły, wtedy dopiero jednorazowo go aktualizujesz i ponawiasz zapytanie, ale znowu musisz na backendzie miec odpowiedni kod błędu i ifa ktory to sprawdza zeby na froncie zlapac "expired". To rozwiazanie bardziej na ruchliwe strony z wieloma zapytaniami i szybkim działaniem.